connectors/sources/notion.py

```diff
@@ -0,0 +1,457 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Notion source module responsible to fetch documents from the Notion Platform."""
+import asyncio
+from copy import copy
+from functools import cached_property, partial
+from urllib.parse import unquote
+
+import aiohttp
+from aiohttp.client_exceptions import ClientResponseError
+from notion_client import APIResponseError, AsyncClient
+from notion_client.helpers import async_iterate_paginated_api
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource, ConfigurableFieldValueError
+from connectors.utils import CancellableSleeps, RetryStrategy, retryable
+
+RETRIES = 3
+RETRY_INTERVAL = 2
+DEFAULT_RETRY_SECONDS = 30
+BASE_URL = "https://api.notion.com"
+MAX_CONCURRENT_CLIENT_SUPPORT = 3
+
+ENDPOINTS = {
+    "search": "/search",
+    "owner": "/users/me",
+}
+
+
+class NotFound(Exception):
+    pass
+
+
+class NotionClient:
+    """Notion API client"""
+
+    def __init__(self, configuration):
+        self._sleeps = CancellableSleeps()
+        self.configuration = configuration
+        self._logger = logger
+        self.notion_secret_key = self.configuration["notion_secret_key"]
+        self.retry_count = self.configuration["retry_count"]
+
+    def set_logger(self, logger_):
+        self._logger = logger_
+
+    @cached_property
+    def _get_client(self):
+        return AsyncClient(
+            auth=self.notion_secret_key,
+            base_url=BASE_URL,
+        )
+
+    @cached_property
+    def session(self):
+        """Generate aiohttp client session.
+
+        Returns:
+            aiohttp.ClientSession: An instance of Client Session
+        """
+        connector = aiohttp.TCPConnector(limit=MAX_CONCURRENT_CLIENT_SUPPORT)
+
+        return aiohttp.ClientSession(
```

Comment:
This doesn't seem to use any auth. Are the URLs we're using the Session for not secured?

Reply:
Yes, these are public URLs as stated in the [API docs](https://developers.notion.com/docs/working-with-files-and-media#retrieving-files-and-media-via-the-notion-api)

Reply:
perfect. Thanks for the explanation!